src/pybind/mgr/dashboard_v2/auth.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+
+import bcrypt
+import cherrypy
+import time
+from cherrypy import tools
+
+class Auth(object):
+    """
+    Provide login and logout actions.
+
+    Supported config-keys:
+
+      | KEY             | DEFAULT | DESCR                                                        |
+      --------------------------------------------------------------------------------------------
+      | username        | None    | Username                                                     |
+      | password        | None    | Password encrypted  using  bcrypt                            |
+      | session-expire  | 1200    | Session will expire after <expires> seconds without activity |
+    """
+
+    SESSION_KEY = '_username'
+    SESSION_KEY_TS = '_username_ts'
+
+    def __init__(self, module):
+        self.module = module
+
+    @cherrypy.expose
+    @cherrypy.tools.allow(methods=['POST'])
+    @tools.json_out()
+    def login(self, username=None, password=None):
+        now = int(time.time())
+        config_username = self.module.get_localized_config('username', None)
+        config_password = self.module.get_localized_config('password', None)
+        hash_password = bcrypt.hashpw(password.encode('utf8'), config_password)
+        if username == config_username and hash_password == config_password:
+            cherrypy.session.regenerate()
+            cherrypy.session[Auth.SESSION_KEY] = username
+            cherrypy.session[Auth.SESSION_KEY_TS] = now
+            return {'username': username}
+        else:
+            cherrypy.response.status = 403
+            return {'detail': 'Invalid credentials'}
+
+    @cherrypy.expose
+    @cherrypy.tools.allow(methods=['POST'])
+    def logout(self):
+        cherrypy.session[Auth.SESSION_KEY] = None
+        cherrypy.session[Auth.SESSION_KEY_TS] = None
+
+    @cherrypy.expose
+    @cherrypy.tools.allow(methods=['POST'])
+    def ping(self):
+        pass
+
+    def check_auth(self):
+        username = cherrypy.session.get(Auth.SESSION_KEY)
+        if not username:
+            raise cherrypy.HTTPError(401,
+                                     'You are not authorized to access that resource')
+        now = int(time.time())
+        expires = int(self.module.get_localized_config('session-expire', 1200))
```

Comment:
Could we make the default session expire time a static variable?

Reply:
Done